test/attr/backDeploy.swift

```diff
@@ -0,0 +1,130 @@
+// RUN: %target-typecheck-verify-swift -parse-as-library
+
+// MARK: - Valid declarations
+
+@available(macOS 12.0, *)
+@_backDeploy(macOS 11.0)
+public func backDeployedTopLevelFunc() {}
```

Comment:
Great work!

Semantically, does this mean we emit the code into client when deploying to macOS 11.0 (inclusively) and before? I wonder what happens if the deployment target is somewhere between 11.0 and 12.0, say 11.4?

Reply:
Never mind, it seems 11.0 is the lower bound of backward deployment, which makes a lot of sense.

Reply:
That's right, it's intended to be a lower bound for inlinability, which is going to help us diagnose when library authors use potentially unavailable APIs from the bodies of back deployed functions.